tilavarauspalvelu/integrations/email/main.py

```diff
@@ -80,81 +137,64 @@ def send_application_received_email(
 
         email_type = EmailType.APPLICATION_RECEIVED
         context = get_context_for_application_received(language=language)
-        send_emails_in_batches_task.delay(
-            recipients=recipients,
-            subject=context["title"],
-            text_content=render_text(email_type=email_type, context=context),
-            html_content=render_html(email_type=email_type, context=context),
-        )
+        email = EmailData.build(recipients, context, email_type)
+        send_emails_in_batches_task.delay(email_data=email)
 
     @staticmethod
-    def send_application_in_allocation_emails() -> None:
-        """Sends emails to applicants when application round has entered the allocation phase."""
-        applications = Application.objects.should_send_in_allocation_email().order_by("created_date")
-        if not applications:
+    def send_application_section_cancelled(
+        application_section: ApplicationSection,
+        *,
+        language: Lang | None = None,
+    ) -> None:
+        """Sends an email that the whole application section was cancelled by the user"""
+        reservation: Reservation | None = application_section.actions.get_last_reservation()
+        if reservation is None:
             return
 
-        recipients_by_language = get_recipients_for_applications_by_language(applications)
-        if not recipients_by_language:
+        recipients = get_reservation_email_recipients(reservation=reservation)
```

Comment:
Should this fetch recipients from the application (`get_application_email_recipients(section.application)`) instead? I think they will be the same with how the series generation works out, but I think that would be more in line with the requirements.

Reply:
Also applies to the staff email.

Reply:
Applies to email contexts too.